webhook/psbinding/reconciler.go

```diff
@@ -361,7 +361,7 @@ func (r *BaseReconciler) ReconcileSubject(ctx context.Context, fb Bindable, muta
 	if r.WithContext != nil {
 		ctx, err = r.WithContext(ctx, fb)
 		if err != nil {
-			return err
+			return apierrs.NewNotFound(gvr.GroupResource(), subject.Name)
```

Comment:
It seems odd to me that we would turn arbitrary `WithContext` errors into "not found".

I get that SinkBinding's implementation of `WithContext`'s main failure mode is "not found", but why not push that inward?

Seems like this path here: https://github.com/knative/eventing/blob/1dfad10d1a3aab6b5c3884f7dd7e57477c81f8bd/pkg/reconciler/sinkbinding/controller.go#L131

Pulling this back into PKG's responsibility, we could make `URIFromDestinationV1` return appropriate apimachinery errors and probably expand this fix to a whole class of errors like this.  WDYT?

Reply:
@mattmoor thanks! yes it was absolutely a somewhat arbitrary decision (based on the return codes catches downward (upward?) in the function call chain looking for `NotFound`.

The reason I put it here, vs, in the sinkbinding controller (as suggested) is when looking at the return type of `WithContext` there's no guarantee of a different `WithContext` implementation wouldn't hit this as well.  While I definitely appreciate the thought (and was my first inclination as well), I went with this approach so that the pkg portion would be more robust in general.  Thoughts?

Reply:
Done here, it just feels to me like a bad assumption about WithContext.  If we can achieve the same by fixing the URI resolver, that'd be my preference for "maximal fixage" :)

Reply:
ok, thanks for the feedback, will close and make the change there.